src/__tests__/starWarsSchema.js

```diff
@@ -26,278 +18,173 @@ import { getFriends, getHero, getHuman, getDroid } from './starWarsData';
-    },
-    secretBackstory: {
-      type: GraphQLString,
-      description: 'Where are they from and how they came to be who they are.',
-      resolve() {
-        throw new Error('secretBackstory is secret.');
-      },
-    },
-  }),
-  interfaces: [characterInterface],
-});
+  """A mechanical creature in the Star Wars universe."""
+  type Droid implements Character {
+    """The id of the droid."""
+    id: String!
+    """The name of the droid."""
+    name: String
+    """The friends of the droid, or an empty list if they have none."""
+    friends: [Character]
+    """Which movies they appear in."""
+    appearsIn: [Episode]
+    """Construction date and the name of the designer."""
+    secretBackstory: String
+    """The primary function of the droid."""
+    primaryFunction: String
+  }
 
-/**
- * The other type of character in Star Wars is a droid.
- *
- * This implements the following type system shorthand:
- *   type Droid : Character {
- *     id: String!
- *     name: String
- *     friends: [Character]
- *     appearsIn: [Episode]
- *     secretBackstory: String
- *     primaryFunction: String
- *   }
- */
-const droidType = new GraphQLObjectType({
-  name: 'Droid',
-  description: 'A mechanical creature in the Star Wars universe.',
-  fields: () => ({
-    id: {
-      type: GraphQLNonNull(GraphQLString),
-      description: 'The id of the droid.',
-    },
-    name: {
-      type: GraphQLString,
-      description: 'The name of the droid.',
-    },
-    friends: {
-      type: GraphQLList(characterInterface),
-      description:
-        'The friends of the droid, or an empty list if they have none.',
-      resolve: droid => getFriends(droid),
-    },
-    appearsIn: {
-      type: GraphQLList(episodeEnum),
-      description: 'Which movies they appear in.',
-    },
-    secretBackstory: {
-      type: GraphQLString,
-      description: 'Construction date and the name of the designer.',
-      resolve() {
-        throw new Error('secretBackstory is secret.');
-      },
-    },
-    primaryFunction: {
-      type: GraphQLString,
-      description: 'The primary function of the droid.',
-    },
-  }),
-  interfaces: [characterInterface],
-});
+  type Query {
+    hero(
+      """If omitted, returns the hero of the whole saga. If provided, returns the hero of that particular episode."""
+      episode: Episode
+    ): Character
+
+    human(
+      """id of the human"""
+      id: String!
+    ): Human
+
+    droid(
+      """id of the droid"""
+      id: String!
+    ): Droid
+  }
+`);
+
+// FIXME: Patch enum defined in SDL, should be possible to do without hacks :(
+const episodeEnum = (StarWarsSchema.getType('Episode'): any);
+episodeEnum.getValue('NEWHOPE').value = 4;
+episodeEnum.getValue('EMPIRE').value = 5;
+episodeEnum.getValue('JEDI').value = 6;
+for (const enumValue of episodeEnum.getValues()) {
+  episodeEnum._valueLookup.set(enumValue.value, enumValue);
+}
```

Comment:
There no way to assign enum values after schema constructed.
@leebyron What do you this about adding one more option to `buidlSchema`/`extendSchema`?
```js
const schema = buildSchema(sdl, {
  extraConfig: {
    Episode: {
      values: {
        NEWHOPE: { value: 4 },
        EMPIRE: { value: 5 },
        JEDI: { value: 6 },
      },
    },
  },
});
```

Reply:
Likewise buildSchema does not provide resolvers or resolveType functions - I think this is probably more appropriate for the graphql-tools which provides additional runtime behavior atop buildSchema

Reply:
I'm implementing this in https://github.com/graphql/graphql-js/pull/1987, it may look like this, if merged:
```js
const schema = buildSchema(sdl, {
  resolvers: {
    Episode: {
      NEWHOPE: 4,
      EMPIRE: 5,
      JEDI: 6,
    },
  },
});
```